src/systems/systems/src/pressurization/mod.rs

```diff
@@ -0,0 +1,952 @@
+    sea_level_pressure: Pressure,
+    destination_qnh: Pressure,
+}
+
+impl AircraftInputsPressurization {
+    fn new() -> Self {
+        Self {
+            eng_1_n1: Ratio::new::<percent>(0.),
+            eng_2_n1: Ratio::new::<percent>(0.),
+            landing_elevation: Length::new::<foot>(0.),
+            sea_level_pressure: Pressure::new::<hectopascal>(1013.),
+            destination_qnh: Pressure::new::<hectopascal>(1013.),
+        }
+    }
+
+    pub fn set_eng_n1(&mut self, eng_1_n1: Ratio, eng_2_n1: Ratio) {
+        self.eng_1_n1 = eng_1_n1;
+        self.eng_2_n1 = eng_2_n1;
+    }
+
+    pub fn set_landing_elev(&mut self, landing_elev: Length) {
+        self.landing_elevation = landing_elev;
+    }
+
+    pub fn set_sea_level_pressure(&mut self, sea_level_pressure: Pressure) {
+        self.sea_level_pressure = sea_level_pressure;
+    }
+
+    pub fn set_destination_qnh(&mut self, dest_qnh: Pressure) {
+        self.destination_qnh = dest_qnh;
+    }
+
+    pub fn eng_1_n1(&self) -> Ratio {
+        self.eng_1_n1
+    }
+
+    pub fn eng_2_n1(&self) -> Ratio {
+        self.eng_2_n1
+    }
+
+    pub fn landing_elev(&self) -> Length {
+        self.landing_elevation
+    }
+
+    pub fn sea_level_pressure(&self) -> Pressure {
+        self.sea_level_pressure
+    }
+
+    pub fn destination_qnh(&self) -> Pressure {
+        self.destination_qnh
+    }
+}
+
+pub struct Pressurization {
+    cpc_1: CabinPressureController,
+    cpc_2: CabinPressureController,
+    outflow_valve: PressureValve,
+    // safety_valve: PressureValve,
+    // rcpu: bool,
+    // is_in_man_mode: bool,
+    // man_mode_duration: Duration,
+    active_system: usize,
+    aircraft_inputs: AircraftInputsPressurization,
+}
+
+impl Pressurization {
+    pub fn new() -> Self {
+        let random = random_number();
+        let mut active: usize = 1;
+        if random % 2 == 0 {
+            active = 2
+        }
+
+        Self {
+            cpc_1: CabinPressureController::new(),
+            cpc_2: CabinPressureController::new(),
+            outflow_valve: PressureValve::new(),
+            // safety_valve: PressureValve::new(),
+            // rcpu:
+            // is_in_man_mode: false,
+            // man_mode_duration: Duration::from_millis(0),
+            active_system: active,
+            aircraft_inputs: AircraftInputsPressurization::new(),
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        press_overhead: &PressurizationOverheadPanel,
+        eng_1_n1: Ratio,
+        eng_2_n1: Ratio,
+    ) {
+        self.aircraft_inputs.set_eng_n1(eng_1_n1, eng_2_n1);
+
+        if !self.is_sys1_active() && !self.is_sys2_active() {
+            self.set_active_system();
+        };
+        if self.is_sys1_active() {
+            self.cpc_1.update(
+                context,
+                &self.outflow_valve,
+                press_overhead,
+                &self.aircraft_inputs,
+            );
```

Comment:
**issue**: I think that only updating the active CPC leads to problems where the state on the inactive CPC will be all wrong when one switches to it. For example, consider CPC's `update_cabin_pressure` setting the `cabin_pressure` based on the previous value which is contained within that particular CPC. Of course cabin pressure itself is something that is a fact or "measured" thing and therefore it maybe shouldn't actually be stored within the CPC.

Reply:
Absolutely right! I didn't consider the situation where CPC's need to swap in-flight (will be captured in future PR), but this wouldn't have worked. I've changed it now so both CPC's are updated for every tick, so the transition from one to the other should be seamless.

Reply:
And regarding where the cabin pressure is stored, according to the documentation I have, both the pressure sensor data and the logic that uses it to control the outflow valve live inside of the CPC. In fact if both CPC's go down, the resolution of the cabin pressure indication to the pilots will go down as the back up system is engaged in that case.

![image](https://user-images.githubusercontent.com/60938089/124368344-40f59080-dc58-11eb-8254-7f724c238e06.png)